packages/flutter/lib/src/material/page.dart

```diff
@@ -221,12 +88,21 @@ class MaterialPageRoute<T> extends PageRoute<T> {
   }
 
   @override
+  bool canTransitionTo(TransitionRoute<dynamic> nextRoute) {
+    if (!(nextRoute is MaterialPageRoute<dynamic>))
+      return false;
+    final MaterialPageRoute<dynamic> nextMaterialPageRoute = nextRoute;
+    // Don't perform outgoing animation if the next route is a fullscreen dialog.
+    return !nextMaterialPageRoute.fullscreenDialog;
```

Comment:
Though I had to implement new APIs on the page route for a sec until I remembered you mentioned it in https://github.com/flutter/flutter/pull/5690#issuecomment-244218142